File: merryChristmas/src/services/deseo.ts
import api from "@/lib/api";

export type Deseo = {
  id_deseo: number;
  descripcion: string;
  estado: "pendiente" | "comprado";
  usuario: {
    id_usuario: number;
    username: string;
  };
}

export const listarDeseos = async () => {
  const { data } = await api.get<Deseo[]>("/deseos");
  return data;
};

export const crearDeseo = async (descripcion: string) => {
  const { data } = await api.post<Deseo>("/deseos", { descripcion });
  return data;
};

export const eliminarDeseo = async (id: number) => {
  const { data } = await api.delete(`/deseos/${id}`);
  return data;
};

export const cambiarEstadoDeseo = async (id: number) => {
  const { data } = await api.patch<Deseo>(`/deseos/${id}/estado`);
  return data;
};

File: backend/app/models/Deseo.py
import enum
from sqlalchemy import Column, Date, Enum, Float, ForeignKey, Integer, String, Text
from sqlalchemy.sql import func
from app.core.database import Base
from sqlalchemy.orm import relationship

class EstadoDeseo(str, enum.Enum):
    pendiente = "pendiente"
    comprado = "comprado"

class Deseo(Base):
    __tablename__='deseos'
    
    id_deseo = Column(Integer, primary_key=True, autoincrement=True, unique=True)
    nombre = Column(String(100), nullable=False)
    precio = Column(Float)
    link = Column(Text)
    descripcion = Column(Text)
    id_usuario = Column(Integer, ForeignKey("usuarios.id_usuario"), nullable=False)
    id_familia = Column(Integer, ForeignKey("familias.id_familia"), nullable=False)
    estado = Column(Enum(EstadoDeseo), default=EstadoDeseo.pendiente)
    fecha_creacion = Column(Date, server_default=func.now())
    
    # Relaciones
    usuario = relationship("Usuario", back_populates="deseos")
    familia = relationship("Familia", back_populates="deseos")